packages/EasyErrorHandler/src/Bridge/Laravel/Translator.php

```diff
@@ -14,19 +15,37 @@ final class Translator implements TranslatorInterface
      */
     private $decorated;
 
-    public function __construct(IlluminateTranslatorInterface $decorated)
+    /**
+     * @var string
+     */
+    private $translationNamespace;
+
+    public function __construct(IlluminateTranslatorInterface $decorated, string $translationNamespace)
     {
         $this->decorated = $decorated;
+        $this->translationNamespace = $translationNamespace;
     }
 
     /**
+     * @param string $message
      * @param mixed[] $parameters
+     *
+     * @return string
      */
     public function trans(string $message, array $parameters): string
     {
         // TODO: rework after upgrading all the illuminate and laravel packages to ^6.0
         $method = \method_exists($this->decorated, 'trans') ? 'trans' : 'get';
 
+        $translation = $this->decorated->{$method}(
+            \implode([$this->translationNamespace, '::', \trim($message)]),
+            $parameters
+        );
+
+        if ($translation !== $message && Str::startsWith($this->translationNamespace . '::', $translation)) {
```

Comment:
Checking if the translated message starts with the translation namespace for early return doesn't make sense, because if it actually translated the message properly then it won't have the namespace in the result.

Also we should try to translate without the namespace first, and default to the namespace because we should have more messages without the namespace than messages with namespace.

Reply:
Thank you, I'll note of this